[Ang Ze Yu] Duke Increments

ang-zeyu opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

ang-zeyu committed:
Add gradle wrapper support

ang-zeyu committed:
gradle setup (ShadowJar, JUnit, checkstyle and application)

ang-zeyu committed:
Setup text-ui-test folder initial files

ang-zeyu committed:
Level 1. Greet, Echo, Exit, with simple test case

ang-zeyu committed:
Level 2. Add, List implementation with simple test case

ang-zeyu committed:
Level 3. Mark as Done & A-Classes extension implementation, with fixed runtest.sh and new test input.

ang-zeyu committed:
Level 4. ToDos, Events, Deadlines and A-inheritance implementation

ang-zeyu committed:
Refactor commands to enums due to larger number of commands in the stage.
Fix runtest classpath for packages and added test case.

ang-zeyu committed:
Change Task package name to more semantic DukeTask
Initial wave of Level-5 : Handle Errors implementations.

ang-zeyu committed:
Refactor reusable common utility functions to new class

ang-zeyu committed:
More extensive level 5 implementation, with minor fix to util function

ang-zeyu committed:
Refactor to use arraylist for storing tasks

ang-zeyu committed:
Level 6. Delete implementation

ang-zeyu committed:
Eliminated redundant check in printTaskArray
A few styling fixes

ang-zeyu committed:
A-Classes task class implementation

ang-zeyu committed:
More styling fixes before level 7, 8 branch

ang-zeyu committed:
Initial save to disk implementation

ang-zeyu committed:
Fixes:
- remove text.NumberFormat phantom import
- erroneous catch position for DukeTaskFileParseException

Changes:
- Added file write implementation
- refactor TaskLetter to TaskType enum, shifted to DukeTask package
- Add TaskType implementation to various Task classes

ang-zeyu committed:
Fixes:
- Unclosed filewriter
- Missing file write info
- Save to disk after mark as done or delete tasks too
- Add missing lineNumber increment
- Missing escape sequence in regex for parsing file
- Move file storage path to main class

Changes:
- Add getters for description and timing to Task class
- Default horizontal line length

ang-zeyu committed:
Changes:
- Repackage all classes and packages into duke, change build gradle and runtest.sh to accomodate
- Change DukeTask package name to task
- Implementation of input time format to save as LocalDateTime in task classes

ang-zeyu committed:
Changes:
- Refactor packages for merge

ang-zeyu committed:
Merge branch 'branch-Level-7'

ang-zeyu committed:
Removed unused method validateTaskPeriod

ang-zeyu committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/task/DeadlineTask.java
#	src/main/java/duke/task/EventTask.java
#	text-ui-test/ACTUAL.TXT
#	text-ui-test/EXPECTED.TXT

Additional Changes:
- Refactor DukeFileUtil (level-7) loadTasksFromDisk method to use newer error handling for time in level-8

ang-zeyu committed:
Groundwork for A-MoreOOP

ang-zeyu committed:
Changes:
- More groundwork : Command abstraction structure, classes, package; Parser full implementation, Main loop refactor

ang-zeyu committed:
Changes:
- refactor Storage classes to use json format instead (add StorageKey enum)
- move Storage parsing logic to Parser
- refactor Storage class to use a relative, recursive parent file directory name

ang-zeyu committed:
Fixes:
- added missing null check for getTaskType and getDoneStatus
- extraneous line divider in bye command
- missing stringbuilder reset after adding k:v pair in Parser
- missing endDate is later than startDate check for eventTask

ang-zeyu committed:
Changes:
- Refactor classes into more packages
- Remove taskType hasTime function and refactor dependents

ang-zeyu committed:
Added:
- DukeUtilTest class

ang-zeyu committed:
Added:
- stricter check for opening brackets in data file when parsing
- StorageParser test class and implementation

ang-zeyu committed:
Changes:
- loosen no leading and trailing whitespace restriction on date time parse functions.
- added TaskUtilTest class and implementation.

ang-zeyu committed:
Added:
- CommandParserTest class and implementation

Fix:
- styling fixes

ang-zeyu committed:
Changes:
- adapt Storage for A-Jar release
- require only directory file name for Storage instead, will try to create fallback directory in working directory

Fixes:
- Removed some unused variables

ang-zeyu committed:
Added:
- find Commands command type
- case for find command type to CommandParser
- FindCommand Command class implementation

Fixes:
- Made number of arguments check for tighter

ang-zeyu committed:
Changes:
- Javadocs for FindCommand class.
- refactor constructor structure to allow more accurate construction of search pattern from command string.

ang-zeyu committed:
Added:
- javadocs for all command classes and DukeExceptions class

ang-zeyu committed:
Changed:
- Added all remaining javadocs
- TaskUtil to abstract

Fix:
- A few javadoc format errors

ang-zeyu committed:
Conformation to coding style.

ang-zeyu committed:
Style Fix:
- Changed all enums (StorageKey, TaskType, Commands) to appropriate constant case.

Changes:
- Convert strings to uppercase before parsing to enum StorageKey or Commands.
- Relaxed parsing for command in so doing to accept both upper and lower case.
- Relaxed parsing storage file's json keys in so doing to accept both upper and lower case.
- Replace enum toString calls with lowercase version (StorageKey, TaskType)

ang-zeyu committed:
Merge branch 'branch-A-JavaDoc'

ang-zeyu committed:
Test file style conformation.

ang-zeyu committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Task.java

ang-zeyu committed:
Merge branch 'branch-Level-9'
- refactor 'find' enum to 'FIND' to comply with style standard

# Conflicts:
#	src/main/java/duke/command/CommandParser.java
#	src/main/java/duke/command/Commands.java
#	src/main/java/duke/command/ListCommand.java

ang-zeyu committed:
Changes:
- Moved invalid description validation from commands to Task
- Added tests for many more classes
- Moved common task test constants to new interface file

Fixes:
- order of task operations preventing size miscalculation if misused
- fix checkstyle.xml line wrapping indent

ang-zeyu committed:
Changes:
- Moved invalid description validation from commands to Task
- Added tests for many more classes
- Moved common task test constants to new interface file

Fixes:
- order of task operations preventing size miscalculation if misused

ang-zeyu committed:
Changes:
- Deleted Ui class, refactored to JavaFX implementations (not yet for tests)
- Refactored main Duke class to use JavaFX event loop structure

ang-zeyu committed:
Changes:
- Deleted UiStub class, refactored to MainWindowStub for tests.

ang-zeyu committed:
Added:
- More test classes

Ellieyee reviewed (commented):
Overall code was very neat and follows the coding standards very well. Tests were also very well written 👍

ang-zeyu committed:
Changed:
- gui general style changes - color, alignment, ...

Fixes:
- Removed unicode smileys since javafx gui dosen't display it well, switched to text emoji
- Changed unicode ticks and crosses back to unicode escape sequences since java fx dosen't display it well
- Some style fixes

ang-zeyu committed:
Changes:
- Checkstyle.xml review :
- Added missing constant style check
- Fixed scoped non-static final variable invalid style errors

ang-zeyu committed:
Merge branch 'branch-A-CheckStyle'

ang-zeyu committed:
Merge branch 'branch-Level-10'
Changes while merging:
- Minor stage style tweak
- Refactored exit process to use javafx's transitions

Fixes while merging:
- Removed unused imports, allow unicode escapes in checkstyle
- Added missing newlines at AddTaskCommand

# Conflicts:
#	src/main/java/duke/ui/Ui.java
#	src/test/java/duke/command/CommandStub.java
#	src/test/java/duke/storage/StorageTest.java
#	src/test/java/duke/task/TaskStub.java

JohnNzj reviewed (commented):
Design of classes was good, made use of abstraction well.
Complies well with java coding standard.
Java docs comments were clear and concise.
Couldn't find any flaws in the design of classes.

ang-zeyu committed:
Fixes:
- extraneous indentation in certain messages for jfx gui
- added '?' quantifier, removing extra initial space requirement for find command empty string search
- removed stage initial style changes

ang-zeyu committed:
Changes:
- Refactor dialogBox to use AnchorPane for easier container placement.
- Move all MainWindow styles to MainWindow.css.

ang-zeyu committed:
Changes :
- Add assertions for various classes
- Enabled assertions when running JUnit tests in build.gradle
- Added last few test classes and implementation

ang-zeyu committed:
Merge pull request #3 from ang-zeyu/branch-A-Assertions

Changes :
- Add assertions for various classes
- Enabled assertions when running JUnit tests in build.gradle
- Added last few test classes and implementation

ang-zeyu committed:
Changes :
- Extracted getCommandType subroutine from parseCommand (CommandParser)
- Improved some test method names
- Refactor duke util int methods implementation and calls to use class variable for indicating and checking failure.
- Added more test cases for duke util concatarray
- Extract getScanner, parseAndStoreTask and createTask methods from loadTasksToList (Storage)
- Extract setDates from initDates (eventTask)
- Extract setNormal/FlippedDialogBoxStyle() from dialogBox constructor
- Add missing newline in delete command message
- Add minimum window width and height for gui

ang-zeyu committed:
Changes :
- Removed unused cleanUpTestDir method ( in place of JUnit's @tempdir )

ang-zeyu committed:
Merge pull request #4 from ang-zeyu/branch-A-CodeQuality

PR Merge A-CodeQuality increment
Also added retroactive tags A-Lambdas and A-Streams which are already used extensively.

ang-zeyu committed:
Fixes :
- Lowered minimum window width
- Revert dialogBoxSize fxml width and height changes to make size respect word wrap

ang-zeyu committed:
branch-A-travis (#5)

Squash merge travis branch

* Added:
- .travis.yml

* Fix permission access for travis

ang-zeyu committed:
Changes :
- Extract exit logic from duke instance to bye command.
- Add secondary constructor to storage to allow custom file name to be used.
- Add deleteAllTasks method to TaskList
- Add archive command and implementation

Fixes :
- Change todo, deadline, event, delete and done command classes to package-private.

ang-zeyu committed:
Added tests for ArchiveCommand

ang-zeyu committed:
Merge pull request #6 from ang-zeyu/branch-C-Archive

Branch-C-archive merge commit

keiteo commented on src/main/java/duke/command/ArchiveCommand.java:
```diff
@@ -0,0 +1,85 @@
+
+        tasks.deleteAllTasks();
+        ui.showMessage("Got it. I've archived all your tasks to disk and wiped the task list!:\n");
+    }
```
Avoid using magic variables - this can be abstracted into the ui class!

keiteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,334 @@
+    }
+
+    /**
+     * Private method to get a scanner from the file path of this storage instance.
```
Javadoc convention - start your description with a verb.

keiteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,334 @@
+    public void saveTasksToDisk(TaskList tasks) throws DukeFileWriteException {
+        try {
+            FileWriter fileWriter = new FileWriter(filePath);
+
```
This method can be further abstracted!

keiteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,334 @@
+            if (status.equalsIgnoreCase("true")) {
+                return true;
+            } else if (status.equalsIgnoreCase("false")) {
+                return false;
```
Try to avoid nested ifelse statements

keiteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,334 @@
+    private void setFilePath() {
+        String workingDir = System.getProperty("user.dir");
+        Path currentDir = Paths.get(workingDir);
+        int recursiveSearchCount = 1;
```
SLAP

keiteo commented on src/main/java/duke/storage/StorageParser.java:
```diff
@@ -0,0 +1,100 @@
+            case ':':
+                isKey = false;
+                break;
+            case ',':
```
Comments to clarify this would be useful

keiteo commented on src/test/java/duke/TestUtils.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    static String generateRandomString(int minChars, int maxChars) {
+        final int minAsciiChar = 33;
+        final int maxAsciiChar = 126;
```
Constant names should be capitalised!

keiteo reviewed (commented):
Good job! Code is generally clean, extensive testing and excellent documentation are demonstrated.

General comments:
- Some of the bigger methods can be further abstracted using SLAP techniques
- Magic variables for ui.sendMessage() can be abstracted into a Ui class instead

ang-zeyu committed:
Changes :
- Extracted create file directory method out of setFilePath (Storage)
- Minor DialogBox style changes
- Add window title
- Added Ui screenshot
- Changed default user image

Fixes :
- Moved deleteAllTasks call position (ArchiveCommand)

ang-zeyu committed:
Changes :
- Added user guide

ang-zeyu committed:
Fixes:
- docs/README formatting fixes

ang-zeyu committed:
Fixes :
- Fix content page anchors for user guide

ang-zeyu committed:
Changes :
- dialogbox style changes
- updated Ui.png
- update release version number to 0.2
- add info message when data directory is created

Fixes :
- gradle javafx cross platform dependencies
- setFilePath searching to beyond root directory causing null exception (Storage)
- javadoc spelling errors
- edit javadoc summary sentences to start with verb where appropriate
